ui/packages/catalog-ui-search/src/main/webapp/react-component/container/workspace-interactions/workspace-interactions.tsx

```diff
@@ -139,6 +141,30 @@ class WorkspaceInteractions extends React.Component<Props, State> {
       wait: true,
     })
   }
+  deletionPrompt = () => {
+    const workspace = store.getWorkspaceById(this.props.workspace.id)
+    const security = new Security(Restrictions.from(workspace))
+
+    if (security.isShared()) {
+      var self = this
+      this.props.listenTo(
+        ConfirmationView.generateConfirmation({
+          prompt:
+            'Are you sure you want to delete this workspace? It has been shared with other users.',
+          no: 'Cancel',
+          yes: 'Delete',
+        }),
+        'change:choice',
+        function(confirmation: any) {
+          if (confirmation.get('choice')) {
+            self.deleteWorkspace()
```

Comment:
If you are binding with `this`, is it still required to use `self`?

Reply:
In all other places where the `ConfirmationView` is used (except for one) they assign `self`. I had someone else look at it too and we couldn't figure out a way around it.

Reply:
so changing it to an arrow function doesn't work?